packages/framework/azure-client/src/test/AzureClient.spec.ts

```diff
@@ -16,47 +17,144 @@ describe("AzureClient", () => {
         );
-
-        const { container } = await resources;
-        assert.notStrictEqual(Object.keys(container.initialObjects), Object.keys(schema.initialObjects));
     });
 
+    /**
+     * Scenario: test when an Azure Client container is created,
+     * it is initially detached.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
     it("Created container is detached", async () => {
         const { container } = await client.createContainer(schema);
         assert.strictEqual(container.attachState, AttachState.Detached, "Container should be detached");
     });
 
-    it("Can attach container", async () => {
+    /**
+     * Scenario: Test attaching a container.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
+    it("can attach a container", async () => {
         const { container } = await client.createContainer(schema);
         const containerId = await container.attach();
+        await new Promise<void>((resolve) => {
+            container.on("connected", () => {
+                resolve();
+            });
+        });
 
         assert.strictEqual(
-            typeof(containerId), "string",
+            typeof (containerId), "string",
             "Attach did not return a string ID",
         );
         assert.strictEqual(
             container.attachState, AttachState.Attached,
             "Container is not attached after attach is called",
         );
+    });
+
+    /**
+     * Scenario: Test if attaching a container twice fails.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
+    it("cannot attach a container twice", async () => {
+        const { container } = await client.createContainer(schema);
+        const containerId = await container.attach();
+        await new Promise<void>((resolve) => {
+            container.on("connected", () => {
+                resolve();
+            });
+        });
+
+        assert.strictEqual(
+            typeof (containerId), "string",
+            "Attach did not return a string ID",
+        );
+        assert.strictEqual(
+            container.attachState, AttachState.Attached,
+            "Container is attached after attach is called",
+        );
         await assert.rejects(
             container.attach(),
             () => true,
-            "Container should not attached twice",
+            "Container should not attach twice",
         );
     });
 
+    /**
+     * Scenario: test if Azure Client can get an existing container.
+     *
+     * Expected behavior: an error should not be thrown nor should a rejected promise
+     * be returned.
+     */
     it("can retrieve existing Azure Fluid Relay container successfully", async () => {
         const { container: newContainer } = await client.createContainer(schema);
```

Comment:
for these cases let's use the `.container` syntax used elsewhere in the test
```suggestion
        const newContainer = (await client.createContainer(schema)).container;
```